### ISSUE TITLE
ACCESS_STORAGE_FOR_INSERTION_NOT_FOUND when creating new user.

### ISSUE THREAD

**Issue by tmikaeld** [ClickHouse/ClickHouse#31487]
Did:
```
CREATE USER api HOST ANY IDENTIFIED WITH sha256_password BY 'qwerty';
```

Got:
```
Code: 514. DB::Exception: Received from 10.10.10.7:9000. DB::Exception: Not found a storage to insert user `api`: Couldn't insert user `api`. Successfully inserted: none. (ACCESS_STORAGE_FOR_INSERTION_NOT_FOUND)

**Comment by den-crane** [comment]
Can you share `select  version()`

`cd /etc/clickhouse-server/; grep local_directory *  -A5`

**Comment by tmikaeld** [comment]
Version: 21.11.4.14

![Image 2021-11-18 at 7 10 24 AM](https://user-images.githubusercontent.com/907195/142361884-f901ac17-3b2c-42ba-9366-a019808f4c31.jpg)

**Comment by tmikaeld** [comment]
Did a full wipe and re-install, works now.

Closing

**Comment by ag2014** [comment]
This bug is also present in version 24.10.1

c2c :) CREATE USER name1 NOT IDENTIFIED

CREATE USER name1 IDENTIFIED WITH no_password


Query id: c05877b9-3007-4e38-abfe-ea7fdd813dd0





Elapsed: 0.003 sec. 



Received exception from server (version 24.10.1):

Code: 514. DB::Exception: Received from localhost:9000. DB::Exception: Could not insert user `name1` because there is no writeable access storage in user directories. (ACCESS_STORAGE_FOR_INSERTION_NOT_FOUND)









server stack trace:

2024.11.06 23:15:05.720766 [ 106599 ] {c05877b9-3007-4e38-abfe-ea7fdd813dd0} <Error> TCPHandler: Code: 514. DB::Exception: Could not insert user `name1` because there is no writeable access storage in user directories. (ACCESS_STORAGE_FOR_INSERTION_NOT_FOUND), Stack trace (when copying this message, always include the lines below):



0. DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000d881bdb

1. DB::Exception::Exception(PreformattedMessage&&, int) @ 0x0000000007f5600c

2. DB::Exception::Exception<String, String const&>(int, FormatStringHelperImpl<std::type_identity<String>::type, std::type_identity<String const&>::type>, String&&, String const&) @ 0x00000000085abd6b

3. DB::MultipleAccessStorage::insertImpl(StrongTypedef<wide::integer<128ul, unsigned int>, DB::UUIDTag> const&, std::shared_ptr<DB::IAccessEntity const> const&, bool, bool, StrongTypedef<wide::integer<128ul, unsigned int>, DB::UUIDTag>*) @ 0x0000000010c1dbfa

4. DB::AccessControl::insertImpl(StrongTypedef<wide::integer<128ul, unsigned int>, DB::UUIDTag> const&, std::shared_ptr<DB::IAccessEntity const> const&, bool, bool, StrongTypedef<wide::integer<128ul, unsigned int>, DB::UUIDTag>*) @ 0x0000000010b38ac9

5. DB::IAccessStorage::insert(std::vector<std::shared_ptr<DB::IAccessEntity const>, std::allocator<std::shared_ptr<DB::IAccessEntity const>>> const&, std::vector<StrongTypedef<wide::integer<128ul, unsigned int>, DB::UUIDTag>, std::allocator<StrongTypedef<wide::integer<128ul, unsigned int>, DB::UUIDTag>>> const&, bool, bool) @ 0x0000000010be0021

6. DB::InterpreterCreateUserQuery::execute() @ 0x0000000012318ff4

7. DB::executeQueryImpl(char const*, char const*, std::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum, DB::ReadBuffer*) @ 0x0000000012146e43

8. DB::executeQuery(String const&, std::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum) @ 0x0000000012142c9a

9. DB::TCPHandler::runImpl() @ 0x0000000013353af4
10. DB::TCPHandler::run() @ 0x000000001336ef78
11. Poco::Net::TCPServerConnection::start() @ 0x0000000016229fe7
12. Poco::Net::TCPServerDispatcher::run() @ 0x000000001622a479
13. Poco::PooledThread::run() @ 0x00000000161f7061
14. Poco::ThreadImpl::runnableEntry(void*) @ 0x00000000161f561d
15. ? @ 0x00007f2ab57d6112
16. ? @ 0x00007f2ab58548f8


Second attempt to launch clickhouse with specific definition to locate the user in a specific path on the local filesystem, details below

+xml config 

      <users>

          <user name="default" default_profile="default">

              <password>password</password>

              <default_database>system</default_database>

              <profiles>

                  <profile name="default">

                      <access_storage>

                          <type>filesystem</type>

                          <path>path_to_my_users_folder_on_ext4_fs</path>

                      </access_storage>

                  </profile>

              </profiles>

          </user>

      </users>



% ./clickhouse server --config ss-config.xml                                                        [1]

Processing configuration file 'ss-config.xml'.

2024.11.07 00:02:33.615969 [ 108027 ] {} <Information> SentryWriter: Sending crash reports is disabled

2024.11.07 00:02:33.698525 [ 108027 ] {} <Information> Application: Starting ClickHouse 24.10.1.1526 (revision: 54491, git hash: 91f403e9139c6f2754e7e30100794d78f2b7466d, build id: 1D4E187BB5FBD8F1147779274A210A803C33D132), PID 108027

2024.11.07 00:02:33.698595 [ 108027 ] {} <Information> Application: starting up

2024.11.07 00:02:33.698612 [ 108027 ] {} <Information> Application: OS name: Linux, version: 6.10.11-amd64, architecture: x86_64

2024.11.07 00:02:33.698661 [ 108027 ] {} <Information> Jemalloc: Value for background_thread set to true (from true)

2024.11.07 00:02:33.705324 [ 108027 ] {} <Information> Application: Available RAM: 2.98 GiB; logical cores: 2; used cores: 2.

2024.11.07 00:02:33.705425 [ 108027 ] {} <Information> Application: Available CPU instruction sets: SSE, SSE2, SSE3, SSSE3, SSE41, SSE42, F16C, POPCNT, PCLMUL, AES, AVX, RDRAND, RDTSCP, XSAVE, OSXSAVE

2024.11.07 00:02:33.710680 [ 108027 ] {} <Information> CgroupsReader: Will create cgroup reader from '/sys/fs/cgroup/user.slice/user-1000.slice/session-2.scope' (cgroups version: v2)

2024.11.07 00:02:33.711129 [ 108027 ] {} <Trace> AsynchronousMetrics: Scanning /sys/class/thermal

2024.11.07 00:02:33.711209 [ 108027 ] {} <Trace> AsynchronousMetrics: Scanning /sys/block

2024.11.07 00:02:33.711424 [ 108027 ] {} <Trace> AsynchronousMetrics: Scanning /sys/devices/system/edac

2024.11.07 00:02:33.711511 [ 108027 ] {} <Trace> AsynchronousMetrics: Scanning /sys/class/hwmon

2024.11.07 00:02:33.713161 [ 108027 ] {} <Information> Application: Shutting down storages.

2024.11.07 00:02:33.713342 [ 108027 ] {} <Trace> Context: Shutting down named sessions

2024.11.07 00:02:33.713480 [ 108027 ] {} <Trace> Context: Shutting down database catalog

2024.11.07 00:02:33.713699 [ 108027 ] {} <Trace> DatabaseCatalog: Shutting down system databases

2024.11.07 00:02:33.713922 [ 108027 ] {} <Trace> Context: Shutting down DDLWorker

2024.11.07 00:02:33.714140 [ 108027 ] {} <Trace> Context: Shutting down caches

2024.11.07 00:02:33.714338 [ 108027 ] {} <Trace> Context: Shutting down AccessControl

2024.11.07 00:02:33.714564 [ 108027 ] {} <Debug> Application: Shut down storages.

2024.11.07 00:02:33.714788 [ 108027 ] {} <Debug> Application: Destroyed global context.

2024.11.07 00:02:33.715162 [ 108027 ] {} <Information> Application: Waiting for background threads

2024.11.07 00:02:33.715304 [ 108027 ] {} <Information> Application: Background threads finished in 0 ms

2024.11.07 00:02:33.716377 [ 108027 ] {} <Error> Application: Code: 76. DB::ErrnoException: Cannot open file /var/lib/clickhouse/status: , errno: 2, strerror: No such file or directory. (CANNOT_OPEN_FILE), Stack trace (when copying this message, always include the lines below):



0. DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000d881bdb

1. DB::Exception::Exception(PreformattedMessage&&, int) @ 0x0000000007f5600c

2. DB::ErrnoException::ErrnoException<String const&>(int, int, FormatStringHelperImpl<std::type_identity<String const&>::type>, String const&) @ 0x000000000d8d2e7d

3. void DB::ErrnoException::throwFromPath<String const&>(int, String const&, FormatStringHelperImpl<std::type_identity<String const&>::type>, String const&) @ 0x000000000d8d2968

4. DB::StatusFile::StatusFile(String, std::function<void (DB::WriteBuffer&)>) @ 0x000000000da95f5e

5. DB::Server::main(std::vector<String, std::allocator<String>> const&) @ 0x000000000da3bcf9

6. Poco::Util::Application::run() @ 0x000000001624a8a6

7. DB::Server::run() @ 0x000000000da372f0

8. Poco::Util::ServerApplication::run(int, char**) @ 0x00000000162536e7

9. mainEntryClickHouseServer(int, char**) @ 0x000000000da34538

10. main @ 0x0000000007f525ee

11. ? @ 0x00007f9c22dd3d68

12. ? @ 0x00007f9c22dd3e25

13. _start @ 0x000000000638a02e

 (version 24.10.1.1526 (official build))

2024.11.07 00:02:33.716636 [ 108027 ] {} <Information> Application: shutting down

2024.11.07 00:02:33.716949 [ 108027 ] {} <Debug> Application: Uninitializing subsystem: Logging Subsystem

2024.11.07 00:02:33.717404 [ 108029 ] {} <Trace> BaseDaemon: Received signal -2

2024.11.07 00:02:33.717534 [ 108029 ] {} <Information> BaseDaemon: Stop SignalListener thread
